Source/misdat.cpp

```diff
@@ -248,4 +248,22 @@ void MissileFileData::LoadGFX()
 	}
 }
 
+void InitMissileGFX(bool loadHellfireGraphics)

```

Comment:
I'd love to not see references to "hellfire" explicitly in the code. The fact that we are introducing yet another one here bothers me a bit.

- Would it be possible to rename this to something more generic that doesn't reference hellfire explicitly? 
- Wouldn't it be better to have 2 loading methods instead of having one with a parameter?
- Why is `MissileSpriteData[]` populated with missiles that may end up not being loadable in the first place? Can't we just iterate through the entire array here, but prevent it from having hellfire missiles added to it somewhere else?

Reply:
Care to elaborate on your downvote @AJenbo ? 

In particular regarding the third bullet above: why do we have entries in `MissileSpriteData` that cannot be loaded? To me, the array should only be populated with entries that can effectively be loaded, and thus the hellfire check should be happening there, and not here.

Reply:
This param was introduced to extract the use of the `gbIsHellfire` global. misdat.cpp does not know of that symbol and I did not want to add another reference to a global that is not strictly necessary.

`MissileSpriteData[]` is initialised when declared instead of being dynamic, changing that is possible but a significant amount of work I really didn't want to get into here 😂